### ISSUE TITLE
f32ToF16 dropped the rounding carry: ~0.03% of values silently halved

### ISSUE THREAD

**Issue by Nehanth** [Nehanth/swarmllm#28]
Fixed in `engine/gguf.js`. Recorded because it explains any earlier quality oddity in split rooms.

`f32ToF16` combined the exponent and the rounded mantissa with `|`:

```js
return sign | (e << 10) | ((m + 0x1000) >> 13);
```

When round-to-nearest carried the mantissa to 0x400, that carry had to propagate into the exponent. With `|` it was dropped whenever the exponent's low bit was already set, so the value came out **exactly halved** (e.g. -1.9999911 -> -1.0).

Measured before the fix: **84 of 300,000 random values (0.028%) were off by 50%**. That is roughly 1.4 corrupted numbers in every 5,120-float activation frame on every hop of a split room, plus occasional halved Q4/Q8 block scales in requantized tensors.

Fix: `+` instead of the second `|`, so the carry propagates. Also hoisted the scratch `Float32Array`/`Uint32Array` out of the function (it is called once per element of every wire frame).

Verified after: all 65,536 representable f16 values round-trip exactly; zero normal-range values above 0.1% error; GPU goldens still pass and the 27B logits moved slightly closer to the reference.

**Comment by Nehanth** [comment]
Fixed in `5212f1faa862c6fec5f7285172743fdf826eebb8`, merged to `main` and deployed to swarmllm.ai.

Verification: all 65,536 representable f16 values round-trip exactly; zero normal-range values above 0.1% error (was 84 in 300,000 off by exactly 50%). All eight GPU suites pass, including the bit-identical speculative-decoding check, and the 27B logits moved slightly closer to the CPU reference.

Left here as a record of what was wrong and for how long, not as open work.
